packages/contracts-periphery/test/contracts/universal/DrippieRedd.spec.ts

```diff
@@ -0,0 +1,260 @@
+import { Contract } from 'ethers'
+import _ from 'lodash'
+import { toRpcHexString } from '@eth-optimism/core-utils'
+
+import { expect } from '../../setup'
+import { deploy } from '../../helpers'
+
+describe('DrippieRedd', () => {
+  const DEFAULT_DRIP_NAME = 'drippity drip drip'
+  const DEFAULT_DRIP_CONFIG = {
+    recipient: '0x' + '11'.repeat(20),
+    amount: hre.ethers.BigNumber.from(1),
+    checkscript: hre.ethers.utils.hexConcat([
+      '0x60', // PUSH1
+      '0x01', // 0x01
+    ]),
+    interval: hre.ethers.BigNumber.from(100),
+    data: '0x',
+  }
+
+  let signer1: SignerWithAddress
+  let signer2: SignerWithAddress
+  before('signer setup', async () => {
+    ;[signer1, signer2] = await hre.ethers.getSigners()
+  })
+
+  let SimpleStorage: Contract
+  let DrippieRedd: Contract
+  beforeEach('deploy contracts', async () => {
+    SimpleStorage = await deploy('SimpleStorage')
+    DrippieRedd = await deploy('DrippieRedd', {
+      signer: signer1,
+      args: [signer1.address],
+    })
+  })
+
+  beforeEach('balance setup', async () => {
+    await hre.ethers.provider.send('hardhat_setBalance', [
+      DrippieRedd.address,
+      toRpcHexString(DEFAULT_DRIP_CONFIG.amount.mul(100000)),
+    ])
+    await hre.ethers.provider.send('hardhat_setBalance', [
+      DEFAULT_DRIP_CONFIG.recipient,
+      '0x0',
+    ])
+  })
+
+  describe('constructor', () => {
+    it('should set the owner', async () => {
+      expect(await DrippieRedd.owner()).to.equal(signer1.address)
+    })
+  })
+
+  describe('create', () => {
+    describe('when called by the owner', () => {
+      it('should create a drip with the given name', async () => {
+        await DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+
+        const drip = await DrippieRedd.drips(DEFAULT_DRIP_NAME)
+        expect(drip.status).to.equal(1)
+        expect(drip.last).to.deep.equal(hre.ethers.BigNumber.from(0))
+        expect(_.toPlainObject(drip.config)).to.deep.include(
+          DEFAULT_DRIP_CONFIG
+        )
+      })
+
+      it('should not be able to create the same drip twice', async () => {
+        await DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+
+        await expect(
+          DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+        ).to.be.revertedWith('DrippieRedd: drip with that name already exists')
+      })
+    })
+
+    describe('when called by not the owner', () => {
+      it('should revert', async () => {
+        await expect(
+          DrippieRedd.connect(signer2).create(
+            DEFAULT_DRIP_NAME,
+            DEFAULT_DRIP_CONFIG
+          )
+        ).to.be.revertedWith('UNAUTHORIZED')
+      })
+    })
+  })
+
+  describe('update', () => {
+    describe('when called by the owner', () => {
+      it('should update the config for the given drip', async () => {
+        await DrippieRedd.create(DEFAULT_DRIP_NAME, DEFAULT_DRIP_CONFIG)
+        await DrippieRedd.update(DEFAULT_DRIP_NAME, {
```

Comment:
flaky: the tx should be waited on.

Reply:
Hardhat is single-threaded so this will always work. Only a potential flake if we run these tests against a non-hardhat network (which I doubt we'll ever do).

Reply:
I feel like its nice to have portable tests, then they can be ran against remote nodes if need be

Reply:
This test suite specifically relies on hardhat-specific behavior, I'd have to do more significant refactoring to make the test portable and even then we haven't really had a usecase for that yet. I think it's ok to leave as-is for now and come back to this if we eventually decide we do want to run unit tests against stuff other than hardhat vm.